File: Module2/ProjectDiscreteLogistic.m
% MCSB Bootcamp
% VStudent - Project 3.1 Discrete logistic

% simulation parameters
ntMax = 1000;
nParam = 1000;

% model parameters to sweep through
rArray = linspace(0.1,2.99,nParam);

% data collection
xSS = zeros(nParam,300);

% loop through parameters
for iParam=1:nParam
    
    % model parameters
    r = rArray(iParam);
    K = 0.6;
   
    x = zeros(1,ntMax);
    
    % initial condition
    x(1) = 0.5;
    
    % loop through time
    for nt=1:ntMax
        x(nt+1) = x(nt) + r*(1-x(nt)/K)*x(nt);
    end
    
    %% data analysis and collection
    if 0
        figure(1); clf; hold on; box on;
        plot(x(1:100), '-ok');
        xlabel('Time');
        ylabel('Population');
    end % finished plotting time series
    
    
    xSS(iParam,:) = x((end-size(xSS,2)+1):end);
    
end % finished loop through parameters

save('xSS.mat', 'xSS');

%% analysis

load('xSS.mat', 'xSS');

% plot parameter-sweep diagram
figure(2); clf; hold on; box on;
plot(rArray,xSS, '.');
ylabel('Population');
xlabel('Growth rate r')

File: Module2/ProjectDiscreteLogistic.py
import numpy as np
import matplotlib.pyplot as plt

# MCSB Bootcamp
# VStudent - Project 3.1 Discrete logistic

# simulation parameters
ntMax = 1000
nParam = 1000

# model parameters to sweep through
rArray = np.linspace(0.1,2.99,nParam)

# data collection
xSS = np.zeros((nParam,300))

# loop through parameters
for iParam in range(nParam):
    
    # model parameters
    r = rArray[iParam]
    K = 0.6
   
    x = np.zeros(ntMax)
    
    # initial condition
    x[0] = 0.5
    
    # loop through time
    for nt in range(ntMax-1): # note that in python array index starts from 0, thus ntMax-1
        x[nt+1] = x[nt] + r*(1-x[nt]/K)*x[nt]
    
    # data analysis and collection
    if False: # equivalent to if 0 in MATLAB
        plt.figure(1)
        plt.plot(x[0:100], 'ok') # '-ok' not recognized in python matplotlib
        plt.xlabel('Time')
        plt.ylabel('Population')
        plt.grid(True) # equivalent to box on in MATLAB
        plt.show() # equivalent to figure(1); clf; hold on;
    
    xSS[iParam,:] = x[-300:] # equivalent to x((end-size(xSS,2)+1):end)

# save the data to a binary file in NumPy `.npy` format
np.save('xSS.npy', xSS)

# analysis

# load the data from the binary file
xSS = np.load('xSS.npy')

# plot parameter-sweep diagram
plt.figure(2)
plt.plot(rArray, xSS, '.') # note that '.' is the equivalent of '.' in MATLAB
plt.ylabel('Population')
plt.xlabel('Growth rate r')
plt.grid(True) # equivalent to box on in MATLAB
plt.show()
